### ISSUE TITLE
Provide API for the cached partition count?

### ISSUE THREAD

**Issue by fxn** [bbalser/elsa#81]
I need to produce messages whose partition can be computed as easily as `rem(user_id, partition_count)`. The `:md5` partitioner could do the job too, need even distribution and order guaranteed by user, but it does unnecessary extra work for my use case.

The `brod` client used internally has the partition counts per topic cached, which is what I need for performance. I am tentatively using this function in my app:

```elixir
def partitions(topic) do
  registry = Elsa.Supervisor.registry(@connection)

  Elsa.Util.with_client(registry, fn client ->
    {:ok, partitions} = :brod_client.get_partitions_count(client, topic)
    partitions
  end)
end
```

Usage is

```elixir
def publish(event) do
  topic = topic(event)
  key = to_string(event.user_id)
  value = Avro.encode(event)
  partition = rem(event.user_id, partitions(topic))

  Elsa.produce(@connection, topic, {key, value}, partition: partition)
end
```

Do you think Elsa could expose that function?

Alternatively, there could be a way to plug custom partitioners, and in that case the partition count would be part of the callback arguments.

Would volunteer a patch if you'd like any of the two options.

**Comment by jeffgrunewald** [comment]
Hey @fxn thanks so much for the issue!
We'd certainly welcome an PRs you have in mind. Our general philosophy has been to favor flexibility in our solutions so what I'm imagining based on your comments above leads me to think we'd probably be more in favor of a customizable partitioner all else being equal but if you've already got the single-use patch more or less ready to go we'd definitely be open to bringing it in.

**Comment by jdenen** [comment]
Closed with #85. Will be released soon.

**Comment by fxn** [comment]
❤️

**Comment by jdenen** [comment]
This has been released as part of `v1.0.0-rc.1`.

**Comment by fxn** [comment]
Thanks a lot!

I tested this RC and all is good in my project.

Only one minor detail is that in the manual code I had before, I had the user ID at hand as an integer, and could compute the partition number directly with it using `rem/2`.

Now, I only have access to the message key, which is must be a string. Therefore, in the partitioner I need to convert it back to an integer.

However, I do not see a way to have both things, since the natural interface for the partitioner is the one in you've implemented.

So, only feedback from trying, but this patch provides the expected API.
